mlir/include/mlir/IR/PatternMatch.h

```diff
@@ -1457,6 +1457,33 @@ class PDLPatternModule {
                                    std::forward<ConstraintFnT>(constraintFn)));
   }
 
+  /// Register a constraint function that produces results with PDL. A
+  /// constraint function with results uses the same type and registry as
+  /// rewrite functions. It may be specified in one of two ways:
+  ///
+  ///   * `void (PatternRewriter &, PDLResultList &, ArrayRef<PDLValue>)`
```

Comment:
Isn't this return value a `LogicalResult`? Because only then we can signal a failure, correct?